libs/wingsdk/src/core/app.ts

```diff
@@ -100,6 +100,12 @@ export abstract class App extends Construct {
     }
   }
 
+  /**
+   * The name of the compilation target.
+   * @internal
+   */
+  public abstract readonly _target: string;
```

Comment:
IMO it'd be useful to have a literal union type for this instead of just string

Reply:
I thought JSII might complain about this but I guess not 🤷 - added